test/test.js

```diff
@@ -2583,3 +2583,146 @@ modulo1(1, 5);`;
     expect(compile(example)).toEqual(expected);
   });
 });
+
+describe('for loops with conditional', () => {
+  it('handles truthy property', () => {
+    const example =
+`
+for value in values when foo
+  loopAction(value);
+`;
+
+    const expected =
+`for (let value of values) {
+  if (foo) {
+    loopAction(value);
+  }
+}`;
+    expect(compile(example)).toEqual(expected);
+  });
+
+  it('handles truthy existential property', () => {
+    const example = `loopAction(value) for value in values when value?`;
+
+    const expected =
+`for (let value of values) {
+  if (value) {
```

Comment:
This is not equivalent to the example (`false` works in the above, but not here).